modtools/create-unit.lua

```diff
@@ -595,22 +444,47 @@ function getRaceCasteIDs(raceStr, casteStr)
 end
 
 local function allocateNewChunk(hist_entity)
-  hist_entity.save_file_id = df.global.unit_chunk_next_id
-  df.global.unit_chunk_next_id = df.global.unit_chunk_next_id+1
-  hist_entity.next_member_idx = 0
-  print("allocating chunk:",hist_entity.save_file_id)
+  if hist_entity then
+    hist_entity.save_file_id = df.global.unit_chunk_next_id
+    df.global.unit_chunk_next_id = df.global.unit_chunk_next_id+1
+    hist_entity.next_member_idx = 0
+    print("allocating chunk:",hist_entity.save_file_id)
+  else
+    local chunkInfo = df.global.world.worldgen
+    chunkInfo.next_unit_chunk_id = df.global.unit_chunk_next_id
+    df.global.unit_chunk_next_id = df.global.unit_chunk_next_id+1
+    chunkInfo.next_unit_chunk_offset = 0
+    print("allocating chunk:",chunkInfo.next_unit_chunk_id)
```

Comment:
would it be useful to differentiate the printed statements so the user can tell which branch was taken?

Reply:
It certainly won't hurt

Reply:
That should do it [6b93420](https://github.com/DFHack/scripts/pull/461/commits/6b93420f22cfefc74aeac53f00ba47ee80fb6257)